### PR TITLE
[GTK] Import 'drmGetFormatName' from libdrm

### PR DESCRIPTION
**Pull request by dpino**
#### f779b8d5fb6fa851f4a08122347e647584f97ad7
<pre>
[GTK] Import &apos;drmGetFormatName&apos; from libdrm
<a href="https://bugs.webkit.org/show_bug.cgi?id=272518">https://bugs.webkit.org/show_bug.cgi?id=272518</a>

Reviewed by Carlos Garcia Campos.

Follow-up to 277506@main. This Changeset fixed the build for Ubuntu
20.04 by checking whether &apos;drmGetFormatName&apos; function is available on
&apos;libdrm&apos;.

The aforementioned function is short and doesn&apos;t have dependencies,
making it easy to import into WebKit codebase. This allows to show DRM
information in some scenarios.

There&apos;s an scenario where the functions &apos;drmGetFormatModifierVendor&apos; and
&apos;drmGetFormatModifierName&apos; are also needed. These two functions are
much harder to import, so a flag is added to skip calling these functions
in case they were not available.

* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::webkitDrmGetFormatName):
(WebKit::renderBufferFormat):
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:

Canonical link: <a href="https://commits.webkit.org/277539@main">https://commits.webkit.org/277539@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/cabe6da3f7f0e643feaa0fa72e4fb9046f2320cf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47956 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27159 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50842 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50637 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44009 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33035 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24647 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/39014 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48538 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/24827 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/41480 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20315 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22303 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6003 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/41249 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/44296 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/43098 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52531 "Built successfully") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/47448 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22999 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/19345 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/46323 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24268 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45369 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25062 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/54945 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6789 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23990 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/11297 "Passed tests") |

### CODE REVIEW

**Comment by obyknovenius** [comment]
I've fixed it in #27263

**Comment by carlosgcampos** [comment]
I think it's better to add the function, it's useful information to show and the function is very simple.

**Comment by dpino** [comment]
I rebase the patch, and change it a bit to incorporate some ideas from Vitaly's patch (I like more to check whether the required function names are available than the version number of libdrm). I'm testing the patch on Debian 11.

File: Source/cmake/OptionsGTK.cmake [hunk 1]
```diff
@@ -325,7 +325,8 @@ if (USE_LIBDRM)
     endif ()
 
     set(CMAKE_REQUIRED_LIBRARIES LibDRM::LibDRM)
-    WEBKIT_CHECK_HAVE_FUNCTION(HAVE_DRM_GET_FORMAT_NAME drmGetFormatName xf86drm.h)
+    WEBKIT_CHECK_HAVE_FUNCTION(HAVE_DRM_GET_FORMAT_MODIFIER_VENDOR drmGetFormatModifierVendor xf86drm.h)
+    WEBKIT_CHECK_HAVE_FUNCTION(HAVE_DRM_GET_FORMAT_MODIFIER_NAME drmGetFormatModifierName xf86drm.h)
```
Comment (obyknovenius) [inline Source/cmake/OptionsGTK.cmake:329]: The same must be done in `Source/cmake/OptionsWPE.cmake`.

**Review by carlosgcampos — APPROVED** [review]

**Pull request merged**